src/main/scala/uk/gov/ons/bi/dataload/exports/HmrcBiCsvExtractor.scala

```diff
@@ -62,20 +34,71 @@ object HmrcBiCsvExtractor {
     log.info(s"BI index file contains ${biData.count} records.")
 
     // Extract the different sets of data we want, and write to output files
-    val legalEntities = getLegalEntities(biData)
+
+    val hmrcOutput = getHMRCOutput(biData, hmrcFile)
+    log.info(s"Writing ${hmrcOutput.count} Legal Entities to $hmrcFile")
+
+    val legalEntities = getLegalEntities(biData, legalFile)
     log.info(s"Writing ${legalEntities.count} Legal Entities to $legalFile")
-    BiCsvWriter.writeCsvOutput(legalEntities, legalFile)
 
-    val vat = getVatExploded(biData)
+    val vat = getVatExploded(biData, vatFile)
     log.info(s"Writing ${vat.count} VAT entries to $vatFile")
-    BiCsvWriter.writeCsvOutput(vat, vatFile)
 
-    val paye =  getPayeExploded(biData)
+    val paye =  getPayeExploded(biData, payeFile)
     log.info(s"Writing ${paye.count} PAYE entries to $payeFile")
-    BiCsvWriter.writeCsvOutput(paye, payeFile)
 
     // Clear cache
     biData.unpersist(false)
   }
 
+  def stringifyArr(stringArr: Column) = concat(lit("["), concat_ws(",", stringArr), lit("]"))
+
+  def getHMRCOutput(df: DataFrame, outputPath: String): DataFrame = {
+    val arrDF = df
```

Comment:
If this dataframe holds vat and paye refs then call it something useful like `vatAndPaye`. `arrDF` is not a great name.